base_event.cpp

```diff
@@ -1,92 +1,153 @@
-}
-
-// display text in output, used for debugging
-void MainFrame::OnPrint(wxCommandEvent& event)
-{
-  std::string text = "treat yourself well";
-  printToOutput(text);
-  return;
-}
-
-// begin calculation
-void MainFrame::OnCalc(wxCommandEvent& event){
-      
-  //std::string filepath = (filepathText->GetValue()).ToStdString();
-//  printToOutput(filepath);
-  
-  // so far only xyz files allowed
-  Ctrl::getInstance()->runCalculation();
-  return;
-}
-
-// browse for atom file
-void MainFrame::OnAtomBrowse(wxCommandEvent& event){
-  std::string filetype = "XYZ files (*.xyz)|*.xyz";
-  OnBrowse(filetype, filepathText);
-  return;
-}
-
-// browse for radius file
-void MainFrame::OnRadiusBrowse(wxCommandEvent& event){
-  std::string filetype = "TXT files (*.txt)|*.txt";
-  OnBrowse(filetype, radiuspathText);
-  return;
-}
-
-// browse (can only be called by another method function)
-void MainFrame::OnBrowse(std::string& filetype, wxTextCtrl* textbox){
-  wxFileDialog openFileDialog
-    (this, 
-     _("Select File"), 
-     "", 
-     "",
-     filetype, 
-     wxFD_OPEN|wxFD_FILE_MUST_EXIST,
-     wxDefaultPosition,
-     wxDefaultSize,
-     "file browser"
-    );
-   
-  // if user closes dialogue
-  if (openFileDialog.ShowModal() == wxID_CANCEL)
-    return;
-    
-  // proceed loading the file chosen by the user;
-  wxFileInputStream input_stream(openFileDialog.GetPath());
-  // if filepath is invalid
-  if (!input_stream.IsOk())
-  {
-    wxLogError("Cannot open file '%s'.", openFileDialog.GetPath());
-    return;
-  }
-  textbox->SetLabel(openFileDialog.GetPath());
-  return;
-}
+#include <wx/wxprec.h>
+
+#ifndef WX_PRECOMP
+#  include <wx/wx.h>
+#endif
+
+#include "base.h"
+#include "controller.h"
+#include <vector>
+#include <utility>
+
+/////////////////
+// EVENT TABLE //
+/////////////////
+
+BEGIN_EVENT_TABLE(MainFrame, wxFrame)
+  EVT_BUTTON(BUTTON_Calc, MainFrame::OnCalc)
+  EVT_BUTTON(BUTTON_Browse, MainFrame::OnAtomBrowse)
+  EVT_BUTTON(BUTTON_Radius, MainFrame::OnRadiusBrowse)
+  EVT_BUTTON(BUTTON_LoadFiles, MainFrame::OnLoadFiles)
+  EVT_GRID_CELL_CHANGING(MainFrame::GridChange)
+END_EVENT_TABLE()
+
+////////////////////////////////
+// METHODS FOR EVENT HANDLING //
+////////////////////////////////
+
+// exit program
+void MainFrame::OnExit(wxCommandEvent& event)
+{
+  this->Close(TRUE);
+}
+
+// display text in output, used for debugging
+void MainFrame::OnPrint(wxCommandEvent& event)
+{
+  std::string text = "treat yourself well";
+  printToOutput(text);
+  return;
+}
+
+// begin calculation
+void MainFrame::OnCalc(wxCommandEvent& event){
+
+  //std::string filepath = (filepathText->GetValue()).ToStdString();
+  //  printToOutput(filepath);
+
+  // Deactivate the Start and Load files buttons during the calculation
+  MainFrame::calcButton->Enable(false);
+  MainFrame::loadFilesButton->Enable(false);
+  // Prevents editing of the atom list grid during the calculation
+  MainFrame::atomListGrid->EnableEditing(false);

```

Comment:
You're calling the functions here, as if they were static functions, i.e., functions of the Class MainFrame. These functions aren't static though, and you're calling them inside an instance of the MainFrame class. This means `MainFrame::` is unnecessary and possibly incorrect.

Also you are calling all these functions again just a few lines later, just with a different argument. This whole block should be contained in a separate function.

Reply:
I have to admit that I don't have a complete understanding of how everything works and should be coded.
I mostly work by trial and error. Make a code that seems to do what I want until it compiles successfully then see if it does the job as intended.
In this case, it did. With this code, for calculations long enough to observed what is happening in the GUI, the atom lists and buttons are deactivated during the calculation.

Reply:
Tbh, the whole View-Controller-Model design is pretty new to me too. I'm just trying to establish some design guide lines to prevent the code from becoming unmaintainable. Maybe I'll make a wiki article about that.

In general, with these comments I'm trying to give you enough context for you to understand why I would have coded it differently. Also, I'm trying to use the correct jargon so you get used to the C.S. terminology. Hope it's understandable!